.github/workflows/ci-cd.yml

```
@@ -1,182 +1,180 @@
+            if [ -f package-lock.json ]; then npm ci; else npm install; fi
+            if npm test --silent; then echo "Node tests passed"; else echo "Node tests failed"; exit 1; fi
+          elif [ -f requirements.txt ] || [ -f setup.py ]; then
+            echo "Detected Python project"
+            python -m pip install -r requirements.txt || true
+            if [ -f pytest.ini ] || [ -d tests ]; then python -m pytest -q; else echo "No pytest tests detected; skipping"; fi
+          else
+            echo "No known dependency manifest found (package.json or requirements.txt). Skipping install/tests.";
+          fi
+
+      - name: SonarQube scan (if configured via secrets)
+        if: ${{ secrets.SONAR_HOST_URL && secrets.SONAR_TOKEN }}
```

actionlint (1.7.12)

[error] context "secrets" is not allowed here. available contexts are "env", "github", "inputs", "job", "matrix", "needs", "runner", "steps", "strategy", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)